core/src/main/scala/com/salesforce/op/stages/impl/feature/SmartTextMapVectorizer.scala

```diff
@@ -138,9 +137,23 @@ class SmartTextMapVectorizer[T <: OPMap[String]]
     val shouldCleanValues = $(cleanText)
     val shouldTrackNulls = $(trackNulls)
 
+    val topKValue = $(topK)
     val allFeatureInfo = aggregatedStats.toSeq.map { textMapStats =>
       textMapStats.keyValueCounts.toSeq.map { case (k, textStats) =>
+        // Estimate the coverage of the top K values
+        val totalCount = textStats.valueCounts.values.sum // total count
+        // Filter by minimum support
+        val filteredStats = textStats.valueCounts.filter { case (_, count) => count >= $(minSupport) }
+        val sorted = filteredStats.toSeq.sortBy(- _._2)
+        val sortedValues = sorted.map(_._2)
+        // Cumulative Count
+        val cumCount = sortedValues.headOption.map(_ => sortedValues.tail.scanLeft(sortedValues.head)(_ + _))
+          .getOrElse(Seq.empty)
+        val coverage = cumCount.lift(math.min(topKValue, cumCount.length) - 1).getOrElse(0L) * 1.0 / totalCount
         val vecMethod: TextVectorizationMethod = textStats match {
+          // If cardinality not respect, but coverage is, then pivot the feature
+          case _ if textStats.valueCounts.size > maxCard && textStats.valueCounts.size > topKValue && coverage > 0 &&
```

Comment:
Even with the comment I am having trouble parsing which conditions you are looking for - can you add a but more?

Reply:
can you just change it to``` coverage > $(coveragePct)``` and get rid of the additional  ```coverage > 0``` check?